.changesets/feat_zelda_feature_cooperative_cancellation.md

```diff
@@ -0,0 +1,22 @@
+## Cooperative Cancellation for Query Planning
+
+This release introduces cooperative cancellation support for query planning operations. This feature allows the router
+to gracefully handle query planning timeouts and cancellations, improving resource utilization.
+Metrics are emitted for cooperative cancellation:
+
+- Records the "outcome" of query planning on the `apollo.router.query_planning.plan.duration` metric.
+- Records the "outcome" of query planning on the `query_planning` span.
+
+### Example
+
+Configuring a timeout in Measure Mode:
+```yaml
+supergraph:
+  query_planning:
+    experimental_cooperative_cancellation:
+      enabled: true
+      mode: measure
+      timeout_in_seconds: 1.0
```

Comment:
We often use human time for timeout or duration. Something like `1s` Do you think we could replicate this for this config ?

Reply:
It would then be called `timeout`.